core/src/main/java/apoc/trigger/TriggerHandler.java

```diff
@@ -57,54 +49,20 @@ private enum Phase {before, after, rollback, afterAsync}
 
     private final AtomicBoolean registeredWithKernel = new AtomicBoolean(false);
 
-    public static final String NOT_ENABLED_ERROR = "Triggers have not been enabled." +
-            " Set 'apoc.trigger.enabled=true' in your apoc.conf file located in the $NEO4J_HOME/conf/ directory.";
-    private final ThrowingFunction<Context, Transaction, ProcedureException> transactionComponentFunction;
-
     public TriggerHandler(GraphDatabaseService db, DatabaseManagementService databaseManagementService,
-                          ApocConfig apocConfig, Log log, GlobalProcedures globalProceduresRegistry,
-                          Pools pools, JobScheduler jobScheduler) {
+                          Log log, Pools pools, JobScheduler jobScheduler) {
         this.db = db;
         this.databaseManagementService = databaseManagementService;
-        this.apocConfig = apocConfig;
         this.log = log;
-        transactionComponentFunction = globalProceduresRegistry.lookupComponentProvider(Transaction.class, true);
         this.pools = pools;
         this.jobScheduler = jobScheduler;
     }
 
-    private boolean isEnabled() {
-        return apocConfig.getBoolean(APOC_TRIGGER_ENABLED);
-    }
-
-    public void checkEnabled() {
-        if (!isEnabled()) {
-            throw new RuntimeException(NOT_ENABLED_ERROR);
-        }
-    }
-
-    private void updateCache() {
-        activeTriggers.clear();
-
-        lastUpdate = System.currentTimeMillis();
-
-        withSystemDb(tx -> {
-            tx.findNodes(SystemLabels.ApocTrigger,
-                    SystemPropertyKeys.database.name(), db.databaseName()).forEachRemaining(
-                    node -> activeTriggers.put(
-                            (String) node.getProperty(SystemPropertyKeys.name.name()),
-                            MapUtil.map(
-                                    "statement", node.getProperty(SystemPropertyKeys.statement.name()),
-                                    "selector", Util.fromJson((String) node.getProperty(SystemPropertyKeys.selector.name()), Map.class),
-                                    "params", Util.fromJson((String) node.getProperty(SystemPropertyKeys.params.name()), Map.class),
-                                    "paused", node.getProperty(SystemPropertyKeys.paused.name())
-                            )
-                    )
-            );
-            return null;
-        });
 
-        reconcileKernelRegistration();
+    public void updateCache() {
+        final boolean hasTriggers = withSystemDb(tx -> getTriggerNodes(db.databaseName(), tx)
+                .hasNext());
```

Comment:
Maybe the `hasTriggers` should be cached as well so that we don't fire a request to the database each time we call this